official/utils/arg_parsers/parsers.py

```diff
@@ -0,0 +1,208 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""Collection of parsers which are shared among the official models.
+
+  The parsers in this module are intended to be used as parents to all arg
+parsers in official models. For instance, one might define a new class:
+
+class ExampleParser(argparse.ArgumentParser):
+  def __init__(self):
+    super(ExampleParser).__init__(parents=[
+      official.utils.arg_parsers.LocationParser(data_dir=True, model_dir=True),
+      official.utils.arg_parsers.DummyParser(use_synthetic_data=True),
+    ])
+
+    self.add_argument(
+      "--application_specific_arg", "-asa",
+      help="This arg is not part of the generic arg offerings."
+    )
+
+"""
+
+import argparse
+
+
+class LocationParser(argparse.ArgumentParser):
+  """Default parser for specifying file locations.
+
+  Args:
+    add_help: Create the "--help" flag. False if class instance is a parent.
+    data_dir: Create a flag for specifying the input data directory.
+    model_dir: Create a flag for specifying the model file directory.
+  """
+
+  def __init__(self, add_help=False, data_dir=False, model_dir=False):
+    super(LocationParser, self).__init__(add_help=add_help)
+
+    if data_dir:
```

Comment:
Since those are not random string and should be a file path, should we do some validation and make sure they exists?

Reply:
I don't think the arg parser is the place for that, since a model may choose some behavior (i.e. prompt for a specific script to be run or download automatically) on a case by case basis.